### PR TITLE
Add non-main package files in DebugInfo.Documents

### LINKED ISSUE #1245: compiler: add not-main-package files to DebugInfo.Documents
At the current moment we always have the only entry `.go` file in `DebugInfo.Documents` even if we use functions from other `.go` packages. Also all methods from debug info have this entry file in sequence points list even if the method itself is not from the entry package or file.

### LINKED ISSUE #1276: Generate proper debug data for `_initialize`
We have proper globals initialization and `init()` runs after #1261, but we're missing correct debug information for this code, this needs to be fixed.

### LINKED ISSUE #1294: Allow to use multiple files in main package
Right now we provide path to a single file in `compile`. We must be able to compile packages with multiple files in it, especially now, when we don't have a single entry point. To maintain compatibility we can switch based on provided path suffix.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1245.
Close #1276.
Close #1294.

Inspect file:
```
❯ ./neogo contract inspect -c --in pkg/compiler/testdata/multi/file1.go
INDEX    OPCODE       PARAMETER
0        INITSSLOT    2                 <<
2        PUSH12
3        STSFLD0
4        RET
5        INITSLOT     1 local, 0 arg
8        PUSH1
9        RET
```

Inspect dir:
```
❯ ./neogo contract inspect -c --in pkg/compiler/testdata/multi/
INDEX    OPCODE       PARAMETER
0        INITSSLOT    3                 <<
2        PUSH12
3        STSFLD0
4        PUSHINT8     1e ("\x1e")
6        STSFLD1
7        RET
8        INITSLOT     1 local, 0 arg
11       PUSH1
12       RET
13       INITSLOT     1 local, 0 arg
16       LDSFLD0
17       LDSFLD1
18       ADD
19       RET
20       INITSLOT     1 local, 0 arg
23       PUSH0
24       CONVERT      20 (" ")
26       RET
```

### CODE REVIEW

File: pkg/compiler/compiler.go [hunk 1]
```diff
@@ -124,21 +148,18 @@ func CompileWithDebugInfo(name string, r io.Reader) ([]byte, *DebugInfo, error)
...
 	}
 	if len(o.Ext) == 0 {
 		o.Ext = fileExt
 	}
-	b, err := ioutil.ReadFile(src)
-	if err != nil {
-		return nil, err
-	}
-	b, di, err := CompileWithDebugInfo(src, bytes.NewReader(b))
+	b, di, err := CompileWithDebugInfo(src, nil)
```
Comment (AnnaShaleva) [inline pkg/compiler/compiler.go:162]: I found one more issue while testing this ability to compile multiple files from main package. We're able to compile non-linked methods, but suppose that we have package `main` with 2 files: `main.go` and `helper.go`:

File `main.go`:
```
package main

func MainFunc() int {
	return HelperFunc()
}
```

File `helper.go`:
```
package main

func HelperFunc() int {
	return 1
}
```

Compiling `main.go` leads to the following error:
```
/home/neospcc/Documents/GitProjects/nspcc-dev/neo-go/examples/main/main.go:4:9: undeclared name: HelperFunc
error while trying to compile smart contract file: couldn't load packages due to errors: main
```

**Comment by AnnaShaleva** [comment]
Perhaps we should also add the example you provided [in the comment](https://github.com/nspcc-dev/neo-go/pull/1295#issue-465412979) to compiler documentation, it would be helpful.

File: pkg/compiler/codegen.go [hunk 2]
```diff
@@ -77,6 +77,11 @@ type codegen struct {
 	// packages contains packages in the order they were loaded.
 	packages []string
 
+	// documents contains paths to all files used by the program.
+	documents []string
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/compiler/codegen.go:81]: It is hard to find out which of them will be used as a sequence points. I don't think it is worth optimizing now. But maybe some heuristic like "do not include interop packages except for `init`" will work.

File: pkg/compiler/compiler.go (hunk 1 above)
Comment (fyrchik) [reply pkg/compiler/compiler.go:162]: This is expected. `go build` behaves similarly:
```
❯ go build main.go
# command-line-arguments
./main.go:4:9: undefined: HelperFunc
```

But I have forgotten to alter `contract compile` CLI to support directories. Will fix.

File: pkg/compiler/analysis.go [hunk 3]
```diff
@@ -163,6 +164,16 @@ func (c *codegen) visitPkg(pkg *types.Package, seen map[string]bool) {
 	c.packages = append(c.packages, pkgPath)
 }
 
+func (c *codegen) fillDocumentInfo() {
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/compiler/analysis.go:167]: Fixed.

**Pull request merged**